test/fail_compilation/pragmainline2.d

```diff
@@ -16,15 +16,15 @@ void foo()
     pragma(inline, false);
     pragma(inline);
     pragma(inline, true);   // this last one will affect to the 'foo'
-    while (0) { }
+    asm { nop; }
 }
 
-pragma(inline, true)   void f1t() { while (0) {} }  // cannot inline
-pragma(inline, false)  void f1f() { while (0) {} }
-pragma(inline)         void f1d() { while (0) {} }
-void f2t() { pragma(inline, true);  while (0) {} }  // cannot inline
-void f2f() { pragma(inline, false); while (0) {} }
```

Comment:
I guess you said about the hack in `ScopeBuffer.resize` method in `std/internal/scopebuffer.d` module.
Fortunately it's still working, because:
1. `WhileStatement` is lowered to `FoStatement` during semantic analysis and could be expanded with this PR, but
2. `BreakStatement` never be inlined because its cost is still `COST_MAX`.